oauth-authenticator-archetype/src/main/resources/archetype-resources/src/main/java/authentication/RedirectUriUtil.java

```diff
@@ -0,0 +1,32 @@
+package ${package}.authentication;
+
+import se.curity.identityserver.sdk.errors.ErrorCode;
+import se.curity.identityserver.sdk.service.ExceptionFactory;
+import se.curity.identityserver.sdk.service.authentication.AuthenticatorInformationProvider;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+
+import static ${package}.descriptor.${pluginName}AuthenticatorPluginDescriptor.CALLBACK;
+
+final class RedirectUriUtil
```

Comment:
Perhaps this type of thing belong in a SDK-utils library of some sort, rather than generating the code for each new project.